pkg/services/authn/authnimpl/sync/user_sync.go

```diff
@@ -305,14 +308,31 @@ func (s *UserSync) updateUserAttributes(ctx context.Context, usr *user.User, id
 		needsUpdate = true
 	}
 
+	// If the user is provisioned, we need to validate the authID
+	if usr.IsProvisioned {
+		s.log.Debug("User is provisioned", "id", id.Email)
```

Comment:
This is something that could be added to the span -> userId + isProvisioned, not the email.

Reply:
Reworked to include both, the span and the debug line